### PR TITLE
Add Terraform Configurations for Minikube and GKE

### PR DESCRIPTION
**Pull request by jfaltermeier**
We want to use Terraform for providing ready-to-use Getting Started setups to new users. 
User may simply use `terraform apply` to create the required infrastructure and get a Theia Cloud sample installation. 
This should also help reproducibility for issues, as we ask for the configuration that was used. 

This PR adds modules for create a Minikube or GKE cluster, installing Theia Cloud + Dependencies, and setting up a Keycloak Realm. 

Those modules are combined to offer a Minikube and a GKE Getting Started configuration. 
We may add more configurations in the future.

### CODE REVIEW

File: terraform/terraform.md [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+Please follow the official installation instruction at <https://developer.hashicorp.com/terraform/downloads?product_intent=terraform>.
+
+If you are unfamiliar with Terraform you may want to have a look at their tutorials avilable at <https://developer.hashicorp.com/terraform/tutorials>.
+
+## Directory Structure
+
+The `modules` directory contains our reuseable terraform modules for creating clusters, installing dependencies via helm, and configuring keycloak. The modules will be used by the actual terraform configurations available in the `configurations` directory.
+
+If you can't use terraform, the `./modules/helm/main.tf` contains the information which helm charts are installed from which helm repository and you may extract the passed values. For and initial Keycloak realm configuration, you may check the values in `./modules/keycloak/main.tf`.
```
Comment (lucas-koehler) [inline terraform/terraform.md:15]:
```suggestion
If you can't use Terraform, the `./modules/helm/main.tf` contains the information which helm charts are installed from which helm repository and you may extract the passed values. For an initial Keycloak realm configuration, you may check the values in `./modules/keycloak/main.tf`.
```

File: terraform/terraform.md [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+
+## Theia Cloud Getting Started
+
+Currently we have Getting Started configuration for the following Clusters/Services:
```
Comment (lucas-koehler) [inline terraform/terraform.md:19]:
```suggestion
Currently, we have Getting Started configurations for the following Clusters/Services:
```

**Review by lucas-koehler — COMMENTED** [review]
Looks pretty good to me already, thanks! I added some inline suggestions to fix minor typos/grammar.

I noticed these empty files:
1. `terraform/modules/helm/outputs.tf`
2. `terraform/modules/keycloak/outputs.tf`
Do we need them?

**Comment by jfaltermeier** [comment]
> Looks pretty good to me already, thanks! I added some inline suggestions to fix minor typos/grammar. I tested the Minikube setup locally and it worked very well. I did not test GKE.
> 
> I noticed these empty files:
> 
> 1. `terraform/modules/helm/outputs.tf`
> 2. `terraform/modules/keycloak/outputs.tf`
>    Do we need them?

I've removed the empty files. 
Also I've fixed an issue with minikube that `eval $(minikube docker-env)` was not working. 

You may test the GKE configuration with our GKE kubernetes project and run `destroy` again after testing. The configuration should produce a valid hostname, so no manual DNS configuration should be required.

**Review by lucas-koehler — APPROVED** [review]
Thanks for the updates:) I also tested the GKE config now and it worked well :)
I also found out that the terraform has a formatter. What do you think of automatically formatting terraform files on save?
For me this worked in VS Code by configuring this in the workspace settings by adding file `.vscode/settings.json` with this content:
```json
{
  "[terraform]": {
    "editor.defaultFormatter": "hashicorp.terraform",
    "editor.formatOnSave": true,
  },
  "[terraform-vars]": {
    "editor.defaultFormatter": "hashicorp.terraform",
    "editor.formatOnSave": true,
  }
}
```

**Comment by lucas-koehler** [comment]
@jfaltermeier Thanks for adding the settings. This is fine for me to be merged then :tada:

**Pull request merged**